wkcuber/image_readers.py

```diff
@@ -266,33 +343,69 @@ def read_dimensions(self, file_name: Path) -> Tuple[int, int]:
 
     def read_channel_count(self, file_name: Path) -> int:
         with TiffFile(file_name) as tif_file:
-            c_count = TiffImageReader.find_count_of_axis(tif_file, "C")
-            s_count = TiffImageReader.find_count_of_axis(tif_file, "S")
-            assert not (
-                c_count > 1 and s_count > 1
-            ), "This file format is currently not supported."
-            if s_count > 1:
-                return s_count
-            else:
-                return c_count
+            return TiffImageReader.find_count_of_axis(
+                tif_file, "C"
+            ) * TiffImageReader.find_count_of_axis(tif_file, "S")
+
+    def read_sample_count(self, file_name: Path) -> int:
+        with TiffFile(file_name) as tif_file:
+            return TiffImageReader.find_count_of_axis(tif_file, "S")
 
     def read_z_slices_per_file(self, file_name: Path) -> int:
         with TiffFile(file_name) as tif_file:
-            i_count = TiffImageReader.find_count_of_axis(tif_file, "I")
-            if i_count > 1:
-                return i_count
-            return TiffImageReader.find_count_of_axis(tif_file, "Z")
+            if self.z_axis_name is None:
+                self._find_right_z_axis_name(tif_file)
+            assert self.z_axis_name is not None, "Z axis name still unclear"
+            return TiffImageReader.find_count_of_axis(tif_file, self.z_axis_name)
 
     def read_dtype(self, file_name: Path) -> str:
         with TiffFile(file_name) as tif_file:
             return tif_file.series[  # pylint: disable=unsubscriptable-object
                 0
             ].dtype.name
 
+    def _find_right_z_axis_name(self, tif_file: TiffFile) -> None:
+        i_count = TiffImageReader.find_count_of_axis(tif_file, "I")
+        z_count = TiffImageReader.find_count_of_axis(tif_file, "Z")
+        q_count = TiffImageReader.find_count_of_axis(tif_file, "Q")
+        if i_count > 1:
+            assert (
+                z_count * q_count == 1
+            ), "Format error, as multiple Z axis names were identified"
+            self.z_axis_name = "I"
+        elif q_count > 1:
+            assert (
+                z_count * i_count == 1
+            ), "Format error, as multiple Z axis names were identified"
+            self.z_axis_name = "Q"
+        else:
+            assert (
+                i_count * q_count == 1
+            ), "Format error, as multiple Z axis names were identified"
+            self.z_axis_name = "Z"
+
 
 class CziImageReader(ImageReader):
     def __init__(self) -> None:
-        self.tile_shape = None
+        self.tile_shape: Optional[List[int]] = None
+        self.channel_count: Optional[int] = None
+        self.width: Optional[int] = None
+        self.height: Optional[int] = None
+        self.existing_axes: Optional[List[int]] = None
+
+    def matches_if_exist(self, index: int, value: int, other_value: int) -> bool:
```

Comment:
I don't quite understand what this method does (in the bigger picture). Maybe there is a better name for it!?